waiter/src/waiter/util/utils.clj

```diff
@@ -270,6 +274,9 @@
                  (str/replace #"\n  $" "\n")))
        :headers (-> headers
                   (assoc-if-absent "content-type" content-type))
+       :waiter/message (-> (or (-> data-map :details :friendly-error-message)
```

Comment:
Why isn't the appropriate message already passed in via the `message` argument?

Reply:
I initially avoided it since `message` is more verbose. But since we have plenty of room with the header value, using `message` now.

The difference is between: (message) `Service description using waiter headers/token improperly configured; concurrency-level must be an integer in the range [1, 10000].` and (friendly-error-message) `concurrency-level must be an integer in the range [1, 10000].`